openerp/addons/base/ir/ir_actions.py

```diff
@@ -150,6 +150,35 @@ def _lookup_report(self, cr, name):
 
         return new_report
 
+    def create_action(self, cr, uid, ids, context=None):
+        """ Create a contextual action for each of the report."""
+        for ir_actions_report_xml in self.browse(cr, uid, ids, context=context):
+            ir_values_id = self.pool['ir.values'].create(cr, SUPERUSER_ID, {
+                'name': ir_actions_report_xml.name,
+                'model': ir_actions_report_xml.model,
+                'key2': 'client_print_multi',
+                'value': "ir.actions.report.xml,%s" % ir_actions_report_xml.id,
+            }, context)
+            ir_actions_report_xml.write({
+                'ir_values_id': ir_values_id,
+            })
+        return True
+
+    def unlink_action(self, cr, uid, ids, context=None):
+        """ Remove the contextual actions created for the reports."""
+        if self.pool['res.users'].has_group(cr, uid, 'base.group_system'):
```

Comment:
Following a conversation with @odony in #1555, but as you just said a `check_access_rights()` makes more sense since you could be have access rights withtout being admin